lib/audited/sweeper.rb

```diff
@@ -18,7 +18,7 @@ def around(controller)
     end
 
     def current_user
-      controller.send(Audited.current_user_method) if controller.respond_to?(Audited.current_user_method, true)
+      lambda { controller.send(Audited.current_user_method) if controller.respond_to?(Audited.current_user_method, true) }
```

Comment:
just store the controller here and don't create a new lambda ?
... then move the remaining code to the sweeper